src/Idris/Elab/Interface.idr

```diff
@@ -284,12 +287,14 @@ elabInterface {vars} fc vis env nest constraints iname params dets mcon body
          let conName_in = maybe (mkCon fc fullIName) id mcon
          -- Machine generated names need to be qualified when looking them up
          conName <- inCurrentNS conName_in
-         let meth_sigs = mapMaybe getSig body -- (FC, RigCount, List FnOpt, Name, (Bool, RawImp))
-         let meth_decls = map (\ (f, c, o, n, b, ty) => (n, c, o, b, ty)) meth_sigs
+         let meth_sigs = mapMaybe getSig body -- (FC, FC, RigCount, List FnOpt, Name, (Bool, RawImp))
+         let meth_decls = map (\ (f, f', c, o, n, b, ty) => (n, c, o, b, ty)) meth_sigs
```

Comment:
This is getting out of hand: we should probably introduce record types for these
different structures we are passing around during elaboration. That would have the
added benefit of documenting what these are meant to be (e.g. the type of `elabMethods`
is modified below to return 2 `FC` instead of one and it's impossible to know what
they correspond to without looking at the implementation).